File: 38-count-and-say/count-and-say.go
func countAndSay(n int) string {
    if n == 1 {
        return "1";
    }

    pre := countAndSay(n - 1);
    var cur strings.Builder;
    count := 1;

    for i := 1; i < len(pre); i++ {
        if pre[i] == pre[i - 1] {
            count++;
        } else {
            cur.WriteString(strconv.Itoa(count));
            cur.WriteByte(pre[i - 1]);
            count = 1;
        }
    }

    cur.WriteString(strconv.Itoa(count));
    cur.WriteByte(pre[len(pre) - 1]);

    return cur.String();
}

File: 38-count-and-say/count-and-say.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1)
            return "1";

        String pre = countAndSay(n - 1);
        StringBuilder cur = new StringBuilder();
        int count = 1;

        for (int j = 1; j < pre.length(); j++) {
            if (pre.charAt(j) == pre.charAt(j - 1)) {
                count++;
            } else {
                cur.append(count).append(pre.charAt(j - 1));
                count = 1;
            }
        }

        cur.append(count).append(pre.charAt(pre.length() - 1));
        pre = cur.toString();

        return pre;
    }
}